### ISSUE TITLE
problem with run_indep_wkflow_with_gnu_par.pl

### ISSUE THREAD

**Issue by hcdenbakker** [iqbal-lab/cortex#7]
perl /home/henk/cortex/scripts/calling/run_indep_wkflow_with_gnu_par.pl --index subclade_index.txt --ref /home/henk/outbryk/species/Listeria/refdata/LmonoJ2_064.fasta --dir_for_ref_objects ref_new_wrapper --vcftools_dir /home/henk/vcftools --outdir out_NW --stampy_bin /home/henk/stampy-1.0.23/stampy.py --kmer 31 --procs 30 --prefix Lmono_NW
Use of uninitialized value $line in scalar chomp at /home/henk/cortex/scripts/calling/build_samples_parallel.pl line 99.
Use of uninitialized value $line in split at /home/henk/cortex/scripts/calling/build_samples_parallel.pl line 100.
Use of uninitialized value $sample in concatenation (.) or string at /home/henk/cortex/scripts/calling/build_samples_parallel.pl line 102.
Use of uninitialized value $sample in concatenation (.) or string at /home/henk/cortex/scripts/calling/build_samples_parallel.pl line 105.
Use of uninitialized value $sample in concatenation (.) or string at /home/henk/cortex/scripts/calling/build_samples_parallel.pl line 121.
ls: cannot access out_NW/*/vcfs/*wk*raw*: No such file or directory

**Comment by rmnorris** [comment]
This was the thing with the memory I think. Check the build log in one of the sample directories and see if it says it ran out of memory

**Comment by iqbal-lab** [comment]
No Rachel, this error happens before it even tries to build a sample - it is failing to read the per-sample index file

**Comment by hcdenbakker** [comment]
Yup, no memory issue, none of the per-sample index files are present, except index_0, and that one is empty

**Comment by iqbal-lab** [comment]
So that is the failure. Is itbecausee run_independent hashardcoded 
Parallel --gnu ?

**Comment by hcdenbakker** [comment]
Are you suggesting to run it without --gnu?

**Comment by hcdenbakker** [comment]
OK, I think i have found the 'bug', I was able to reproduce exactly the same error while writing a bash-based wrapper; parallel --gnu does not seem to like it if you try to put a variable in the {1..".$num_samples."} part of the script ( {1..$num_samples} in bash). Maybe this is linux distro related? Solved this in the bash wrapper by replacing {1.. $num_samples} with $(seq $num_samples); I don't know if there is a perl equivalent for seq.

**Comment by iqbal-lab** [comment]
Bloody parallels is distro dependent. But if you remove --gnu, you might now find the script works

**Comment by hcdenbakker** [comment]
Nope, removing --gnu does not do the trick for Ubuntu

**Comment by hcdenbakker** [comment]
OK, may have solved it. Created a pull request.
